homeassistant/components/rest/utils.py

```diff
@@ -23,5 +23,5 @@ def render_templates(tpl_dict: dict[str, Template] | None):
     rendered_items = {}
     for item_name, template_header in tpl_dict.items():
         if (value := template_header.async_render()) is not None:
-            rendered_items[item_name] = value
+            rendered_items[item_name] = str(value)
```

Comment:
Is this conversion compatible with all the uses of this tool?

Reply:
Yes. In fact numbers are rendered as native types, that do not have an encoding mehod. Where e.d. headers are a number this will ensure that the header can be encoded. The only use of pvoutput is a REST GET request that results in a sensor.

Reply:
This is not correct. Parsing something to be a Python type and then back to a string will not yield the same results. That's why we have introduced `parse_result` parameter for template rendering. 

Fix here https://github.com/home-assistant/core/pull/61269